portal-bridge/src/beacon_bridge.rs

```diff
@@ -298,21 +311,41 @@ impl BeaconBridge {
     async fn serve_light_client_finality_update(
         api: ConsensusApi,
         portal_clients: Arc<Vec<HttpClient>>,
-    ) -> anyhow::Result<()> {
+        finalized_slot: u64,
+    ) -> anyhow::Result<u64> {
         let data = api.get_lc_finality_update().await?;
         let update: Value = serde_json::from_str(&data)?;
         let update: LightClientFinalityUpdateCapella =
             serde_json::from_value(update["data"].clone())?;
         info!(
-            "Got lc finality update for slot {:?}",
-            update.attested_header.beacon.slot
+            finalized_slot = %update.finalized_header.beacon.slot,
+            "Got LightClientFinalityUpdate",
         );
+
+        let new_finalized_slot = update.finalized_header.beacon.slot;
+
+        match new_finalized_slot.cmp(&finalized_slot) {
+            Ordering::Equal => {
+                // We already gossiped the latest finality updated with the same finalized slot, no need to serve it again.
+                return Ok(finalized_slot);
+            }
+            Ordering::Less => {
+                // if we panic here, it is a bug
+                bail!("System clock went backwards: Expected finalized slot is less than known finalized slot");
```

Comment:
The clock isn't necessarily the culprit here, right? I'm not sure we can even really speculated what when wrong. Maybe as simple as stating what should never happen, which is that a finalized block should never get rewound and "unfinalized".

```suggestion
                bail!("Consensus client must not unwind finalized block, but: New finalized slot is less than known finalized slot");
```